### PR TITLE
chore: bump Node.js to v22.9.0

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Upgrade Node.js to v22.9.0.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Upgrades Node.js to v22.9.0.

### CODE REVIEW

File: spec/lib/events-helpers.ts
Comment (VerteDinde) [inline spec/lib/events-helpers.ts:23]: Does this change a return type here? 👀

File: lib/node/asar-fs-wrapper.ts
Comment (VerteDinde) [inline lib/node/asar-fs-wrapper.ts:862]: Oh no, did they change how to package JSON is read again? 😭

**Review by VerteDinde — APPROVED** [review]
Overall it looks great! Left a few comments on what I thought could be potential breaking or unexpected changes, but am not going to block on them

File: .github/workflows/pipeline-segment-electron-test.yml
Comment (codebytere) [reply .github/workflows/pipeline-segment-electron-test.yml:179]: @VerteDinde this triages https://github.com/nodejs/node/issues/55584 - it's a false positive because not everything is instrumented with ASAN (in this case simdjson is not) and there will be a better fix upstream but in the meantime this prevent a crash for what's actually a valid read (see linked issue). Once a real fix is merged we can backport that and re-enable this

File: patches/sqlite/fix_rename_sqlite_win32_exports_to_avoid_conflicts_with_node_js.patch
Comment (codebytere) [reply patches/sqlite/fix_rename_sqlite_win32_exports_to_avoid_conflicts_with_node_js.patch:10]: Correct - this would be invisible to end users!

File: lib/node/asar-fs-wrapper.ts
Comment (codebytere) [reply lib/node/asar-fs-wrapper.ts:862]: Yep - this should match existing functionality though for all asar-related reads

File: lib/node/asar-fs-wrapper.ts
Comment (codebytere) [inline lib/node/asar-fs-wrapper.ts:844]: Refs https://github.com/nodejs/node/pull/50322 - needed to enable correct package.json reads for asar

File: patches/node/build_enable_perfetto.patch
Comment (codebytere) [inline patches/node/build_enable_perfetto.patch:23]: `src/tracing/node_trace_buffer.h` and `src/tracing/node_trace_writer.h` don't work with Perfetto which is why they are removed

File: patches/node/build_enable_perfetto.patch
Comment (codebytere) [inline patches/node/build_enable_perfetto.patch:86]: See https://github.com/electron/electron/pull/44281/files#r1824973801

File: patches/node/fix_handle_boringssl_and_openssl_incompatibilities.patch
Comment (codebytere) [inline patches/node/fix_handle_boringssl_and_openssl_incompatibilities.patch:27]: Upstreamed at https://github.com/nodejs/node/pull/55425

File: lib/node/asar-fs-wrapper.ts
Comment (samuelmaddock) [reply lib/node/asar-fs-wrapper.ts:844]: Can we add a comment in the code to reference this?

File: patches/node/.patches
Comment (samuelmaddock) [inline patches/node/.patches:44]: great work cleaning this up!

File: patches/node/build_enable_perfetto.patch
Comment (samuelmaddock) [reply patches/node/build_enable_perfetto.patch:23]: Does "don't work" mean it doesn't compile? I assume we lose certain node internal events being added to perfetto traces by removing this.

File: lib/node/asar-fs-wrapper.ts
Comment (codebytere) [reply lib/node/asar-fs-wrapper.ts:844]: @samuelmaddock we can but essentially every function in this file serves the same purpose - would you just want the file generally to be a little better documented?

File: patches/node/build_enable_perfetto.patch
Comment (codebytere) [reply patches/node/build_enable_perfetto.patch:23]: `v8::platform::tracing::TraceBuffer`, which `NodeTraceBuffer` inherits from, is not a part of Perfetto at all: https://github.com/nodejs/node/blob/5ab3140dfbf0dfa33a66a6af62942b4290f370c1/deps/v8/BUILD.gn#L6838-L6848

(same applies to `NodeTraceWriter`)

File: patches/node/build_compile_with_c_20_support.patch
Comment (codebytere) [reply patches/node/build_compile_with_c_20_support.patch:21]: It shouldn't - this fixes a file split change but won't change anything more than was already changed in https://github.com/electron/electron/pull/43555!

File: spec/fixtures/native-addon/external-ab/binding.cc
Comment (codebytere) [inline spec/fixtures/native-addon/external-ab/binding.cc:2]: This used to be necessary due to a conflict in `#include <node_buffer.h>` and `#include <node_api.h>` born of duplicate declarations, and isn't anymore as the duplicates got refactored out.

File: spec/api-utility-process-spec.ts
Comment (codebytere) [inline spec/api-utility-process-spec.ts:513]: This fails in ASAN but importantly does _not_ cause a crash (the test just fails), so I'm disabling it for now to unblock the roll. There are no issues in non-ASAN builds.

File: patches/node/win_almost_fix_race_detecting_esrch_in_uv_kill.patch
Comment (codebytere) [inline patches/node/win_almost_fix_race_detecting_esrch_in_uv_kill.patch:1]: This addresses https://github.com/nodejs/node/issues/51766 - which we were seeing in CI on Windows.

**Review by jkleinsc — APPROVED** [review]

**Review by jkleinsc — COMMENTED** [review]
API LGTM

**Review by ckerr — APPROVED** [review]
API LGTM

**Pull request merged**
